src/rdfTypes/GeometryInfoHelpersImpl.h

```diff
@@ -219,68 +219,70 @@ enum class AnyGeometryMember : uint8_t {
   MULTIPOINT
 };
 
-// Convert an instance of the custom container type `AnyGeometry` from
-// `pb_util` to the variant used here.
-inline ParsedWkt anyGeometryToVariant(const AnyGeometry<CoordType>& geom) {
+// Forward declare so the compiler can see the function for cyclic recursion.
+double computeMetricLengthAnyGeom(const AnyGeometry<CoordType>& geom);
+
+// Helper to implement the computation of metric length for the different
+// geometry types. This is used for visiting the `ParsedWkt` variant in
+// `computeMetricLength` below as well as for computing the length for
+// geometries from the dynamic container class `AnyGeometry`.
+struct MetricLengthVisitor {
+  template <typename T>
+  auto operator()(const T& geom) const {
+    if constexpr (SameAsAny<T, Point<CoordType>, MultiPoint<CoordType>>) {
+      return 0.0;
+    } else if constexpr (std::is_same_v<T, Line<CoordType>>) {
+      return latLngLen<CoordType>(geom);
+    } else if constexpr (std::is_same_v<T, Polygon<CoordType>>) {
+      return computeMetricLengthPolygon(geom);
```

Comment:
does this have to be a separate function still? It also can be alternatively an overload of the operator() or inlined directly here?